pkg/kube/object_patch/validation.go

```diff
@@ -0,0 +1,199 @@
+    properties:
+      apiVersion:
+        type: string
+      kind:
+        type: string
+      name:
+        type: string
+
+type: object
+additionalProperties: false
+properties:
+  operation: {}
+  namespace: {}
+  subresource: {}
+  apiVersion: {}
+  kind: {}
+  name: {}
+  object: {}
+  jsonPatch: {}
+  jqFilter: {}
+  mergePatch: {}
+
+oneOf:
+- allOf:
+  - properties:
+      operation:
+        type: string
+        enum: ["Create", "CreateOrUpdate"]
+  - "$ref": "#/definitions/common"
+  - "$ref": "#/definitions/create"
+- allOf:
+  - properties:
+      operation:
+        type: string
+        enum: ["Delete", "DeleteInBackground", "DeleteNonCascading"]
+  - "$ref": "#/definitions/common"
+  - "$ref": "#/definitions/delete"
+- allOf:
+  - oneOf:
+    - properties:
+        operation:
+          type: string
+          enum: ["JQPatch"]
+        jqFilter:
+          type: string
+          minimum: 1
+    - properties:
+        operation:
+          type: string
+          enum: ["MergePatch"]
+        mergePatch:
+          type: object
+          minProperties: 1
+    - properties:
+        operation:
+          type: string
+          enum: ["JSONPatch"]
+        jsonPatch:
+          type: array
+          minItems: 1
+          items:
+          - type: object
+            required: ["op", "path", "value"]
+            properties:
+              op:
+                type: string
+                minLength: 1
+              path:
+                type: string
+                minLength: 1
+              value: {}
+  - "$ref": "#/definitions/common"
+  - "$ref": "#/definitions/patch"
+`,
+}
+
+var SchemasCache = map[string]*spec.Schema{}
+
+// GetSchema returns loaded schema.
+func GetSchema(name string) *spec.Schema {
+	if s, ok := SchemasCache[name]; ok {
+		return s
+	}
+	if _, ok := Schemas[name]; !ok {
+		return nil
+	}
+
+	// ignore error because load is guaranteed by tests
+	SchemasCache[name], _ = LoadSchema(name)
+	return SchemasCache[name]
+}
+
+// LoadSchema returns spec.Schema object loaded from yaml in Schemas map.
+func LoadSchema(name string) (*spec.Schema, error) {
+	yml, err := swag.BytesToYAMLDoc([]byte(Schemas[name]))
+	if err != nil {
+		return nil, fmt.Errorf("yaml unmarshal: %v", err)
+	}
+	d, err := swag.YAMLToJSON(yml)
+	if err != nil {
+		return nil, fmt.Errorf("yaml to json: %v", err)
+	}
+
+	s := new(spec.Schema)
+
+	if err := json.Unmarshal(d, s); err != nil {
+		return nil, fmt.Errorf("json unmarshal: %v", err)
+	}
+
+	err = spec.ExpandSchema(s, s, nil /*new(noopResCache)*/)
+	if err != nil {
+		return nil, fmt.Errorf("expand schema: %v", err)
+	}
+
+	return s, nil
+}
+
+// See https://github.com/kubernetes/apiextensions-apiserver/blob/1bb376f70aa2c6f2dec9a8c7f05384adbfac7fbb/pkg/apiserver/validation/validation.go#L47
+func ValidateOperationSpec(obj interface{}, s *spec.Schema, rootName string) (multiErr error) {
+	if s == nil {
+		return fmt.Errorf("validate config: schema is not provided")
```

Comment:
I don't think that the word "config" is correct for object patch validation (frankly, errors look like copypasted from another place). In my opinion, it would better to use specific terms like "object patch validation" or something like this.